android/java/org/chromium/chrome/browser/externalnav/BraveExternalNavigationHandler.java

```diff
@@ -28,14 +31,21 @@ public BraveExternalNavigationHandler(ExternalNavigationDelegate delegate) {
 
     @Override
     public OverrideUrlLoadingResult shouldOverrideUrlLoading(ExternalNavigationParams params) {
+        String originalUrl = params.getUrl().getSpec();
         if (isWalletProviderOverride(params)) {
-            String originalUrl = params.getUrl().getSpec();
             String url = originalUrl.replaceFirst("^rewards://", "brave://rewards/");
             GURL browserFallbackGURL = new GURL(url);
             if (params.getRedirectHandler() != null) {
                 params.getRedirectHandler().setShouldNotOverrideUrlLoadingOnCurrentRedirectChain();
             }
             return OverrideUrlLoadingResult.forNavigateTab(browserFallbackGURL, params);
+        } else if (originalUrl.equalsIgnoreCase("chrome://adblock/")) {
```

Comment:
@tapanmodh we should also check for brave://adblock.

Reply:
we are getting chrome://adblock only when we type brave://adblock in the url